src/redis/driver_impl.go

```diff
@@ -204,22 +265,37 @@ func (c *clientImpl) NumActiveConns() int {
 }
 
 func (c *clientImpl) PipeAppend(pipeline Pipeline, rcv interface{}, cmd, key string, args ...interface{}) Pipeline {
-	return append(pipeline, radix.FlatCmd(rcv, cmd, key, args...))
+	// Combine key and args into a single slice
+	allArgs := make([]interface{}, 0, 1+len(args))
+	allArgs = append(allArgs, key)
+	allArgs = append(allArgs, args...)
+	return append(pipeline, radix.FlatCmd(rcv, cmd, allArgs...))
 }
 
 func (c *clientImpl) PipeDo(pipeline Pipeline) error {
-	if c.implicitPipelining {
+	ctx := context.Background()
+	if c.useExplicitPipeline {
```

Comment:
WIP: Even without `useExplicitPipeline` in Redis Cluster, it may be possible to improve performance by pipelining INCR and EXPIRE for the same key in a single pipeline. I'll verify this and follow up with a commit.